File: hibernate-query-language/src/main/java/io/github/robertomike/hefesto/constructors/ConstructOrderImplementation.java
package io.github.robertomike.hefesto.constructors;

import io.github.robertomike.hefesto.actions.Order;

public class ConstructOrderImplementation extends ConstructOrder {
    public String construct() {
        StringBuilder ordersQuery = new StringBuilder();

        items.forEach((value) -> {
            if (!ordersQuery.isEmpty()) {
                ordersQuery.append(", ");
            }

            ordersQuery.append(apply(value));
        });

        if (!items.isEmpty()) {
            ordersQuery.insert(0, "order by ");
        }

        return ordersQuery.toString();
    }

    public String apply(Order value) {
        return value.getField() + " " + value.getSort().name();
    }
}

File: hibernate-query-language/src/main/kotlin/io/github/robertomike/hefesto/constructors/ConstructOrderImplementation.kt
package io.github.robertomike.hefesto.constructors

import io.github.robertomike.hefesto.actions.Order

class ConstructOrderImplementation : ConstructOrder() {
    fun construct(): String {
        val ordersQuery = StringBuilder()

        items.forEach { value ->
            if (ordersQuery.isNotEmpty()) {
                ordersQuery.append(", ")
            }

            ordersQuery.append(apply(value))
        }

        if (items.isNotEmpty()) {
            ordersQuery.insert(0, "order by ")
        }

        return ordersQuery.toString()
    }

    fun apply(value: Order): String {
        return "${value.field} ${value.sort.name}"
    }
}
